extensions/ql-vscode/test/vscode-tests/no-workspace/query-history/query-history-scrubber.test.ts

```diff
@@ -13,77 +13,61 @@ import {
   TWO_HOURS_IN_MS,
 } from "../../../../src/common/time";
 import { mockedObject } from "../../utils/mocking.helpers";
+import { DirResult } from "tmp";
 
-describe("query history scrubber", () => {
-  const now = Date.now();
+const now = Date.now();
+// We don't want our times to align exactly with the hour,
+// so we can better mimic real life
+const LESS_THAN_ONE_DAY = ONE_DAY_IN_MS - 1000;
 
+describe("query history scrubber", () => {
   let deregister: vscode.Disposable | undefined;
-  let mockCtx: vscode.ExtensionContext;
-  let runCount = 0;
```

Comment:
Ok from a bit of research it appears I was right and variables like this are not reset between tests. So the reason it worked is that only the first of the two tests in this file references this variable, and tests within a file are run sequentially in order (at least by default).

The reason it was failing like this is that we have jest configured to retry failed tests, and on the second try it would fail because `runCount` wasn't reset. The [full log](https://github.com/github/vscode-codeql/actions/runs/5750276179/job/15586716425), which I'm embarrassed I didn't read more fully earlier on, is:
```
PASS test/vscode-tests/no-workspace/query-history/variant-analysis-history.test.ts
 LOGGING RETRY ERRORS  query history scrubber should not throw an error when the query directory does not exist
 RETRY 1 

    TreeError [codeQLQueryHistory] Data tree node not found: [object Object]

      at we.A (../../../vscode-file:/vscode-app/home/runner/work/vscode-codeql/vscode-codeql/extensions/ql-vscode/.vscode-test/vscode-linux-x64-1.77.3/resources/app/out/vs/workbench/workbench.desktop.main.js:554:18061)
      at we.reveal (../../../vscode-file:/vscode-app/home/runner/work/vscode-codeql/vscode-codeql/extensions/ql-vscode/.vscode-test/vscode-linux-x64-1.77.3/resources/app/out/vs/workbench/workbench.desktop.main.js:554:17723)
      at Me.reveal (../../../vscode-file:/vscode-app/home/runner/work/vscode-codeql/vscode-codeql/extensions/ql-vscode/.vscode-test/vscode-linux-x64-1.77.3/resources/app/out/vs/workbench/workbench.desktop.main.js:2526:13663)
      at g.j (../../../vscode-file:/vscode-app/home/runner/work/vscode-codeql/vscode-codeql/extensions/ql-vscode/.vscode-test/vscode-linux-x64-1.77.3/resources/app/out/vs/workbench/workbench.desktop.main.js:1586:70284)

 RETRY 2 

    expect(received).toBe(expected) // Object.is equality

    Expected: 0
    Received: 16

      69 |     await wait();
      70 |     // "Should not have called the scrubber"
    > 71 |     expect(runCount).toBe(0);
         |                      ^
      72 |
      73 |     jest.advanceTimersByTime(ONE_HOUR_IN_MS - 1);
      74 |     await wait();

      at Object.<anonymous> (query-history/query-history-scrubber.test.ts:71:22)
```
So you can see the first failure had a different reason, and then the subsequent failures were because of `runCount`.

So with this in mind I'm pretty confident this PR will at least make the test behave consistently on retries. It appears we may have another reason for test failure, but if that's not consistent then it may be handled by the automatic retries.